app/new_password_required.go

```diff
@@ -0,0 +1,44 @@
+package app
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go/aws"
+	cognito "github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+)
+
+// NewPasswordRequired handles set new password.
+func (a *App) NewPasswordRequired(w http.ResponseWriter, r *http.Request) {
+	r.ParseForm()
+
+	username := r.Form.Get("username")
+	password := r.Form.Get("password")
+	session := r.Form.Get("session")
+
+	params := map[string]*string{
+		"USERNAME":     aws.String(username),
+		"NEW_PASSWORD": aws.String(password),
+	}
+	// Compute secret hash based on client secret.
+	if a.AppClientSecret != "" {
+		secretHash := computeSecretHash(a.AppClientSecret, username, a.AppClientID)
+
+		params["SECRET_HASH"] = aws.String(secretHash)
+	}
+	chall := &cognito.RespondToAuthChallengeInput{
+		ChallengeName:      aws.String("NEW_PASSWORD_REQUIRED"),
```

Comment:
Forgot one thing - this should be a constant IMO just like I mentioned [here](https://github.com/br4in3x/golang-cognito-example/pull/8#discussion_r645684504).